### PR TITLE
Improve and simplify YAML test runner

### PR DESCRIPTION
**Pull request by fpagnoux**
Connected to openfisca/openfisca-france#397
Connected to openfisca/openfisca-france#722
Depends on openfisca/openfisca-dummy-country#2

### CODE REVIEW

File: openfisca_core/taxbenefitsystems.py [hunk 1]
```diff
@@ -232,6 +235,38 @@ def load_extension(self, extension):
...
+        Exemple:
+
+        >>> self.apply_reform('openfisca_france.reforms.inversion_revenus')
+
+        """
+        from reforms import Reform
+        try:
+            reform_package, reform_name = reform_path.rsplit('.', 1)
+        except ValueError:
+            raise ValueError(u'`{}` does not seem to be a path pointing to a reform. A path looks like `some_country_package.reforms.some_reform.`'.format(reform_path))
```
Comment (cbenz) [inline openfisca_core/taxbenefitsystems.py]: Missing `.encode('utf-8')`

File: openfisca_core/taxbenefitsystems.py [hunk 2]
```diff
@@ -232,6 +235,38 @@ def load_extension(self, extension):
...
+        try:
+            reform_module = importlib.import_module(reform_package)
+        except ImportError:
+            raise ValueError(u'Could not import `{}`.'.format(reform_package))
```
Comment (cbenz) [inline openfisca_core/taxbenefitsystems.py]: Missing `.encode('utf-8')` – I feel so sorry for that

File: openfisca_core/taxbenefitsystems.py [hunk 3]
```diff
@@ -232,6 +235,38 @@ def load_extension(self, extension):
...
+        reform = getattr(reform_module, reform_name, None)
+        if reform is None:
+            raise ValueError(u'{} has no attribute {}'.format(reform_package, reform_name))
```
Comment (cbenz) [inline openfisca_core/taxbenefitsystems.py]: Missing `.encode('utf-8')`

File: openfisca_core/taxbenefitsystems.py [hunk 4]
```diff
@@ -232,6 +235,38 @@ def load_extension(self, extension):
...
+        if not issubclass(reform, Reform):
+            raise ValueError(u'`{}` does not seem to be a valid Openfisca reform.'.format(reform_path))
```
Comment (cbenz) [inline openfisca_core/taxbenefitsystems.py]: Missing `.encode('utf-8')`

File: openfisca_core/taxbenefitsystems.py (hunk 1 above)
Comment (fpagnoux) [inline openfisca_core/taxbenefitsystems.py]: 👍 

I promised  myself I'd be more careful about this, but I forgot as I just moved this code around

**Comment by fpagnoux** [comment]
Args, tests were passing with a local editable install of `dummy-country`, but not with a regular install.

**Pull request merged**